src/phoenix/experimental/evals/templates/template.py

```diff
@@ -81,3 +92,20 @@ def _parse_variables(self) -> List[str]:
                 variables.append(variable_name)
 
         return variables
+
+
+def _deduce_binary_rails(text: str) -> Tuple[str, str]:
+    match = re.search(_rails_pattern, text)
+    if match is not None:
+        rails = cast(Tuple[str, str], match.groups())
+        logger.info(f"deducing binary rails from text: {', '.join(rails)}")
+        return rails
+    raise ValueError(
+        "binary rails cannot be deduced from text. rails must be specified at initialization"
+    )
+
+
+_rails_pattern = re.compile(
+    r"""response +(?:should|must|shall|has +to|needs +to) +be\b.*?[`"'](\w+)[`"'] +or +[`"'](\w+)[`"']""",  # noqa: E501
+    re.IGNORECASE,
+)
```

Comment:
I'd vote for asking for `rails` from the user rather than trying to infer them from the prompt. There's gonna be edge cases where this is going to give an unexpected result and cause a very confusing bug for the user.